### PR TITLE
Support other filesystem types

### PR DESCRIPTION
**Pull request by jonasasx**
Resolves https://github.com/metal-stack/csi-driver-lvm/issues/13
For example xfs added

### CODE REVIEW

File: pkg/lvm/lvm.go
Comment (majst01) [inline pkg/lvm/lvm.go:164]: Check for second condition can be inside upper if

**Review by majst01 — CHANGES_REQUESTED** [review]
Thanks for this contribution, only some small nits

File: examples/csi-storageclass-linear-xfs.yaml
Comment (majst01) [inline examples/csi-storageclass-linear-xfs.yaml:11]: some sort of validation is required what fsType(s) are supported, imagine someone specifies btrfs here

Comment (jonasasx) [reply examples/csi-storageclass-linear-xfs.yaml:11]: Firstly, fsType is csi field: https://pkg.go.dev/github.com/container-storage-interface/spec/lib/go/csi#VolumeCapability_MountVolume

Secondary, btrfs will work also if the driver will be installed in the container.
Range of the values is limited by fs drivers in the container.

File: pkg/lvm/lvm.go
Comment (jonasasx) [reply pkg/lvm/lvm.go]: fixed

File: pkg/lvm/lvm.go
Comment (jonasasx) [reply pkg/lvm/lvm.go]: fixed

File: examples/csi-storageclass-linear-xfs.yaml
Comment (jonasasx) [reply examples/csi-storageclass-linear-xfs.yaml:11]: In other csi drivers (for example https://github.com/vmware/cloud-director-named-disk-csi-driver.git) I don't see any validation of fsType.

Comment (majst01) [reply examples/csi-storageclass-linear-xfs.yaml:11]: This might be the case, but it would be great if the consumer of csi-driver-lvm will see a proper error message that the given fsType is not supported.

This can be done for example with https://pkg.go.dev/os/exec#LookPath to `mkfs.fsType`

Comment (jonasasx) [reply examples/csi-storageclass-linear-xfs.yaml:11]: And when the validation must be occured? Via ValidatingWebhookConfiguration? On provisioning? On attaching?

Comment (majst01) [reply examples/csi-storageclass-linear-xfs.yaml:11]: At least the error should be so informative that the user can get an idea why his PV is not provisioned. If i remember correctly, errors raised even from the nodeserver are shown in `kubectl describe pv,pvc`. If this is achieved i am fine.

Comment (jonasasx) [reply examples/csi-storageclass-linear-xfs.yaml:11]: Now I have tested the current code. And I see in pod's events:

`MountVolume.SetUp failed for volume "pvc-f1793398-f0ba-440b-bf9b-da55f7f1bc8d" : rpc error: code = Unknown desc = unable to mount lv: unable to format lv:pvc-f1793398-f0ba-440b-bf9b-da55f7f1bc8d err:exec: "mkfs.btrfs": executable file not found in $PATH output:`

Isn't it enough?

Comment (majst01) [reply examples/csi-storageclass-linear-xfs.yaml:11]: Sure.

**Review by majst01 — APPROVED** [review]

**Comment by majst01** [comment]
I added the @mwennrich as reviewer as well, lets wait for his approval

**Comment by majst01** [comment]
first #78 must be merged, to make CI pass.

**Comment by majst01** [comment]
Can you please add a `bats` (in tests/bats ) to test with xfs as well

**Comment by jonasasx** [comment]
Due testing I found an Issue with old volumes signatures.
If we create volume where another volume had been before, blkid returns xfs_external_log type. So I added force formatting in this case.

**Comment by majst01** [comment]
You must silent this warning:

G204: Subprocess launched with a potential tainted input or cmd arguments (gosec)

**Comment by jonasasx** [comment]
@majst01 help with failed pipeline, please.

**Comment by majst01** [comment]
> @majst01 help with failed pipeline, please.

This is because you PRs from outside are not able to log in into our registry for push.

From my side this is fine now !

**Review by majst01 — APPROVED** [review]

**Comment by jonasasx** [comment]
When this feature will be released?

**Comment by Gerrit91** [comment]
We made a release now and container images are also published on ghcr. You can use this deployment chart as well if you want: https://github.com/metal-stack/helm-charts/releases/tag/v0.3.27

Thanks for your contribution!

**Pull request merged**
